File: hmwk_02.cs
// Amey Jayant Puranik
// 1001451390
// 2018-02-28

using System;
using System.IO;
using System.Text.RegularExpressions;

public class hmwk_02
{
  static public void processToken( string token ) {
   
       Regex intger = new Regex(@"^[0][dD][0-9]+$");
       Regex ID = new Regex(@"^[_a-mN-Z][n-zA-M_0-9.]*$");
       Regex deci = new Regex(@"^[0-9]+\.[0-9]+$");
       Regex Hexnt = new Regex(@"^[0][xX][a-fA-F0-9]+$");
       Regex EFP = new Regex(@"^((^[0-9][0-9]*[eE]+[-+]+[0-9]+$)||(^[0-9]*\.[0-9]*[eE]*[+-]*[0-9]*$))$");
      if (intger.IsMatch(token))
 { 
   Console.WriteLine( ">" + token + "< Matches Integer." );
          
 }

    else if (ID.IsMatch(token))
  { 
   Console.WriteLine( ">" + token + "< Matches ID." );
          
 }
   else if (deci.IsMatch(token))
  { 
   Console.WriteLine( ">" + token + "< Matches FP." );
          
 } 
 else if (EFP.IsMatch(token))
  { 
   Console.WriteLine( ">" + token + "< Matches EFP." );
          
   }  
 else if (Hexnt.IsMatch(token))
  { 
   Console.WriteLine( ">" + token + "< Matches HEXINT." );
          
  } 
   
   else
      { 
   Console.WriteLine( ">" + token + "< Does not Match." );
          
      }  
      
    
  }

  static public void Main( string[] args )
  {
    Console.WriteLine( "processing tokens from " + args[ 0 ] + " ..." );

    foreach ( string line in File.ReadAllLines( args[ 0 ] ) ) {
      foreach ( string token in line.Split( (char []) null, StringSplitOptions.RemoveEmptyEntries ) ) {
        processToken( token );
      }
    }
  }
}

File: hmwk_02.cc
//Amey J Puranik
//axp1390
//03/27/2018

#include <iostream>
#include <fstream>
#include <regex>

using namespace std;

void processToken( string token )
{
regex Intt ("[0][dD][0-9]+");
regex FP("[0-9]+\\.[0-9]+");
regex ID("[_a-mN-Z][n-zA-M_0-9]*");
regex Hexnt("[0][xX][a-fA-F0-9]+");
regex EP ("[0-9][0-9]*[eE+-]+[0-9]+||[0-9]*\\.[0-9]*[eE+-]*[0-9]*");


if (regex_match (token, Intt))
      cout << ">" << token << "< Matches INT.\n";

else if (regex_match (token, FP))
     cout << ">" << token << "< Matches FP.\n";

else if (regex_match (token, ID))
     cout << ">" << token << "< Matches ID.\n";

else if (regex_match (token, Hexnt))
     cout << ">" << token << "< Matches HEXINT.\n";

else if (regex_match (token, EP))
     cout << ">" << token << "< Matches EFP.\n";
else
    cout << ">" << token << "< Does Not Match.\n";

}

int main( int argc, char *argv[] )
{
  if ( argc > 1 ) {
    cout << "processing tokens from " << argv[ 1 ] << " ...\n";

    ifstream inputFile;
    string   token;

    inputFile.open( argv[ 1 ] );

    if ( inputFile.is_open() ) {
      while ( inputFile >> token ) {
        processToken( token );
      }
      inputFile.close();
    } else {
      cout << "unable to open " << argv[ 1 ] << "?\n";
    }
  } else {
    cout << "No input file specified.\n";
  }
}
